mesonbuild/build.py

```diff
@@ -634,11 +637,12 @@ def link(self, target):
             if hasattr(t, 'held_object'):
                 t = t.held_object
             if not isinstance(t, (StaticLibrary, SharedLibrary)):
-                raise InvalidArguments('Link target is not library.')
+                raise InvalidArguments('Link target {!r} is not library.'.format(t.name))
             if isinstance(self, SharedLibrary) and isinstance(t, StaticLibrary) and not t.pic:
-                raise InvalidArguments("Can't link a non-PIC static library into a shared library")
+                msg = "Can't link non-PIC static library {!r} into shared library {!r}. ".format(t.name, self.name)
+                msg += "Use the 'pic' option to static_library to build with PIC."
```

Comment:
We don't automagically enable options anywhere else, and I think it's a bad idea to do this in general. For instance, people might install other static libraries to the system without knowing that they have to enable `pic` for them to be usable.